files/common/lib/open-iscsi/unique-name-check.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+#
+# Copyright (c) 2021 by Delphix. All rights reserved.
+#
+
+#
+# This script generates a unique initiator name based on the Delphix Engine UUID
+#
+
+PATH=/bin
+
+EIDFILE="/etc/engine-uuid"
+BOSMGMT="/opt/delphix/server/bin/bos_mgmt"
```

Comment:
This adds an implicit dependency on delphix-virtualization, which seems wrong given that delphix-virtualization depends on delphix-platform. Does this indicate that perhaps the UUID logic should live in delphix-platform rather than delphix-virtualization?

Reply:
Great point. The `unique-name-check` was designed to only run if an engine id is supplied.

Given that the IQN is now tied to the engine UUID, does it make sense to change to a `delphix-iscsi-init` service and move it into `dlpx-app-gate`?

Reply:
I think if we want to tie the IQN to the engine UUID, then that would make sense to me.

Have you considered putting this logic into cloud-init and have it tied to the instance?

Reply:
From what I observed in cloud init, there isn't an instance-id when running on ESX

I could also decouple this from the engine-uuid and directly use something like `dmidecode -s system-uuid`

Reply:
We have added some code in cloud-init to the NoCloud DataSource (used by ESX) that checks for the system-uuid to determine if the "instance" has changed.